plugins/search-backend-node/src/IndexBuilder.ts

```diff
@@ -33,12 +35,22 @@ type IndexBuilderOptions = {
 export class IndexBuilder {
   private collators: Record<string, CollatorEnvelope>;
   private decorators: Record<string, DocumentDecorator[]>;
+  private searchEngine: SearchEngine;
   private logger: Logger;
 
   constructor({ logger }: IndexBuilderOptions) {
     this.collators = {};
     this.decorators = {};
     this.logger = logger;
+    this.searchEngine = new LunrSearchEngine({ logger });
+  }
+
+  setSearchEngine(searchEngine: SearchEngine) {
```

Comment:
is there a use case for dynamically switching this out during runtime? what happens to the indexed data when you do? i thought this would be a parameter to the constructor instead.

Reply:
There is just a use case of being able to configure what search engine to use and that the Lunr one should be the default (you should not have to configure that). 

So you mean having something like this: 

```
const searchEngine = new LunrSearchEngine({ logger })
const indexBuilder = new IndexBuilder({ logger, searchEngine });
```
but still have the getter method to be able to pass it to the createRouter?

```
return await createRouter({
    engine: indexBuilder.getSearchEngine(),
    logger,
});
```

Reply:
I guess. Yeah just giving stuff to constructors in general.